x-pack/legacy/plugins/security/public/views/management/edit_role/components/privileges/es/index_privileges.test.tsx

```diff
@@ -10,59 +10,47 @@ import { RoleValidator } from '../../../lib/validate_role';
 import { IndexPrivilegeForm } from './index_privilege_form';
 import { IndexPrivileges } from './index_privileges';
 
-test('it renders without crashing', () => {
-  const props = {
-    role: {
-      name: '',
-      elasticsearch: {
-        cluster: [],
-        indices: [],
```

Comment:
@legrego -- passing in an empty `indices` array was causing a rejected promise. As a result, the component never mounted correctly in this test (see `index_privileges.test.tsx.snap`).

Instead of adding a check to `index_privileges.tsx` to change the behavior and allow that test to pass, I opted to change the props and pass in a non-empty `indices` array. I assume this is intended functionality, but please take a look and correct it if I made the wrong call.

Reply:
Thanks for taking a look Joe! I still think it's valuable to test that the component can render without any `indices` provided, as this is a common real-world use-case. Taking a look at the component itself, I think the offending code is here:

https://github.com/elastic/kibana/blob/0a527c658c50572d66a71a62cb47844edea7ab4a/x-pack/legacy/plugins/security/public/views/management/edit_role/components/privileges/es/index_privileges.tsx#L157-L164

Since we are calling `this.setState` in a callback, it is possible that we won't call this until after the component has unmounted. I believe that's what's happening in the test case. I think the "proper" fix is to track the mounted status of this component, and only call `this.setState` if the component is still mounted.

Reply:
OK, for future reference the problem was actually Enzyme not being able to handle a component that renders as an empty array or null. Wrapping the `index_privileges_form` component in a `Fragment` fixed the problem!